// dummy-tree-shake-test/Input.js
'use client';

import { jsx as _jsx, jsxs as _jsxs } from 'react/jsx-runtime';

export function Input({ label, error, className = '', style = {}, ...props }) {
  const containerStyles = {
    display: 'flex',
    flexDirection: 'column',
    gap: '4px'
  };

  const labelStyles = {
    fontSize: '14px',
    fontWeight: '500',
    color: '#374151',
    margin: '0'
  };

  const inputStyles = {
    padding: '8px 12px',
    border: `1px solid ${error ? '#ef4444' : '#d1d5db'}`,
    borderRadius: '6px',
    fontSize: '14px',
    outline: 'none',
    transition: 'border-color 0.2s ease',
    ...style
  };

  const errorStyles = {
    fontSize: '12px',
    color: '#ef4444',
    margin: '0'
  };

  return _jsxs('div', {
    style: containerStyles,
    className: className,
    children: [
      label && _jsx('label', { style: labelStyles, children: label }),
      _jsx('input', { style: inputStyles, ...props }),
      error && _jsx('p', { style: errorStyles, children: error })
    ]
  });
}

export default Input;
