// dummy-tree-shake-test/Modal.js
'use client';

import { jsx as _jsx, jsxs as _jsxs } from 'react/jsx-runtime';

export function Modal({ isOpen, onClose, title, children, className = '', ...props }) {
  if (!isOpen) return null;

  const overlayStyles = {
    position: 'fixed',
    top: 0,
    left: 0,
    right: 0,
    bottom: 0,
    backgroundColor: 'rgba(0, 0, 0, 0.5)',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    zIndex: 1000
  };

  const modalStyles = {
    backgroundColor: 'white',
    borderRadius: '8px',
    padding: '24px',
    minWidth: '400px',
    maxWidth: '90vw',
    maxHeight: '90vh',
    overflow: 'auto',
    boxShadow: '0 10px 25px rgba(0, 0, 0, 0.1)'
  };

  const headerStyles = {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: '16px'
  };

  const titleStyles = {
    fontSize: '20px',
    fontWeight: '600',
    color: '#1e293b',
    margin: '0'
  };

  const closeButtonStyles = {
    background: 'none',
    border: 'none',
    fontSize: '24px',
    cursor: 'pointer',
    color: '#64748b',
    padding: '0',
    width: '24px',
    height: '24px'
  };

  return _jsx('div', {
    style: overlayStyles,
    onClick: onClose,
    children: _jsxs('div', {
      style: modalStyles,
      className: className,
      onClick: (e) => e.stopPropagation(),
      ...props,
      children: [
        _jsxs('div', {
          style: headerStyles,
          children: [
            title && _jsx('h2', { style: titleStyles, children: title }),
            _jsx('button', { style: closeButtonStyles, onClick: onClose, children: '×' })
          ]
        }),
        children
      ]
    })
  });
}

export default Modal;
