// react-tree-shake-test/Button.js
'use client';

import { jsx as _jsx } from 'react/jsx-runtime';

export function Button({ children, onClick, variant = 'primary', disabled = false, ...props }) {
  // TREE_SHAKE_MARKER: Button component included
  const componentId = 'react-tree-shake-test-Button-v1.0.0';
  const baseStyles = {
    padding: '8px 16px',
    border: 'none',
    borderRadius: '4px',
    cursor: disabled ? 'not-allowed' : 'pointer',
    fontSize: '14px',
    fontWeight: '500',
    transition: 'all 0.2s ease'
  };

  const variants = {
    primary: {
      backgroundColor: disabled ? '#94a3b8' : '#3b82f6',
      color: 'white'
    },
    secondary: {
      backgroundColor: disabled ? '#f1f5f9' : '#e2e8f0',
      color: disabled ? '#94a3b8' : '#334155'
    },
    danger: {
      backgroundColor: disabled ? '#fca5a5' : '#ef4444',
      color: 'white'
    }
  };

  const styles = {
    ...baseStyles,
    ...variants[variant],
    opacity: disabled ? 0.6 : 1
  };

  return _jsx('button', {
    style: styles,
    onClick: disabled ? undefined : onClick,
    disabled: disabled,
    'data-test-id': componentId,
    ...props,
    children: children
  });
}

export default Button;
