// react-tree-shake-test/Badge.js
'use client';

import { jsx as _jsx } from 'react/jsx-runtime';

export function Badge({ children, variant = 'default', size = 'medium', className = '', style = {}, ...props }) {
  // TREE_SHAKE_MARKER: Badge component included
  const componentId = 'react-tree-shake-test-Badge-v1.0.0';
  const baseStyles = {
    display: 'inline-flex',
    alignItems: 'center',
    fontWeight: '500',
    borderRadius: '9999px',
    textAlign: 'center',
    whiteSpace: 'nowrap'
  };

  const sizes = {
    small: {
      fontSize: '11px',
      padding: '2px 8px'
    },
    medium: {
      fontSize: '12px',
      padding: '4px 12px'
    },
    large: {
      fontSize: '14px',
      padding: '6px 16px'
    }
  };

  const variants = {
    default: {
      backgroundColor: '#f1f5f9',
      color: '#475569'
    },
    success: {
      backgroundColor: '#dcfce7',
      color: '#166534'
    },
    warning: {
      backgroundColor: '#fef3c7',
      color: '#92400e'
    },
    error: {
      backgroundColor: '#fee2e2',
      color: '#991b1b'
    },
    info: {
      backgroundColor: '#dbeafe',
      color: '#1e40af'
    }
  };

  const styles = {
    ...baseStyles,
    ...sizes[size],
    ...variants[variant],
    ...style
  };

  return _jsx('span', {
    style: styles,
    className: className,
    'data-test-id': componentId,
    ...props,
    children: children
  });
}

export default Badge;
