// react-tree-shake-test/Card.js
'use client';

import { jsx as _jsx, jsxs as _jsxs, Fragment as _Fragment } from 'react/jsx-runtime';

export function Card({ children, title, subtitle, className = '', style = {}, ...props }) {
  // TREE_SHAKE_MARKER: Card component included
  const componentId = 'react-tree-shake-test-Card-v1.0.0';
  const cardStyles = {
    backgroundColor: 'white',
    border: '1px solid #e2e8f0',
    borderRadius: '8px',
    padding: '16px',
    boxShadow: '0 1px 3px 0 rgba(0, 0, 0, 0.1)',
    ...style
  };

  const headerStyles = {
    marginBottom: title || subtitle ? '12px' : '0'
  };

  const titleStyles = {
    fontSize: '18px',
    fontWeight: '600',
    color: '#1e293b',
    margin: '0 0 4px 0'
  };

  const subtitleStyles = {
    fontSize: '14px',
    color: '#64748b',
    margin: '0'
  };

  return _jsxs('div', {
    style: cardStyles,
    className: className,
    'data-test-id': componentId,
    ...props,
    children: [
      (title || subtitle) && _jsxs('div', {
        style: headerStyles,
        children: [
          title && _jsx('h3', { style: titleStyles, children: title }),
          subtitle && _jsx('p', { style: subtitleStyles, children: subtitle })
        ]
      }),
      children
    ]
  });
}

export default Card;
